.flake8

```diff
@@ -1,3 +1,20 @@
 [flake8]
 # Matching black's default
 max-line-length = 88
+
+extend-ignore =
+    # TODO: Use per-file-ignores to iterate towards complete docstrings
+    D100,  # Missing docstring in public module
+    D101,  # Missing docstring in public class
+    D102,  # Missing docstring in public method
+    D103,  # Missing docstring in public function
+    D104,  # Missing docstring in public package
+    D107,  # Missing docstring in __init__
```

Comment:
Would we want to leave any of these ignored? For example, D104 and D107?

Reply:
101 and 107 are conflicting in my opinion. We'd want one or the other, but I tend to use `flake8-docstrings` with no configuration in other projects so I suspect some of these are ignored by default, I just don't remember which.

`D100` wouldn't be a problem in our tests if we didn't need to have `tests/__init__.py` and there's little harm in documenting test modules and what tests go in those modules since it makes it easier for new contributors to group tests where we think they belong

Reply:
Removing `tests/__init__.py` doesn't seem to change the results of `D100`. Here's the results with it deleted:

```
$ flake8 --extend-ignore=D1 --select=D100 twine tests
twine/package.py:1:1: D100 Missing docstring in public module
twine/auth.py:1:1: D100 Missing docstring in public module
twine/_installed.py:1:1: D100 Missing docstring in public module
twine/cli.py:1:1: D100 Missing docstring in public module
twine/wininst.py:1:1: D100 Missing docstring in public module
twine/utils.py:1:1: D100 Missing docstring in public module
twine/repository.py:1:1: D100 Missing docstring in public module
twine/__main__.py:1:1: D100 Missing docstring in public module
twine/wheel.py:1:1: D100 Missing docstring in public module
twine/commands/check.py:1:1: D100 Missing docstring in public module
twine/commands/upload.py:1:1: D100 Missing docstring in public module
twine/commands/register.py:1:1: D100 Missing docstring in public module
tests/test_repository.py:1:1: D100 Missing docstring in public module
tests/test_utils.py:1:1: D100 Missing docstring in public module
tests/conftest.py:1:1: D100 Missing docstring in public module
tests/test_package.py:1:1: D100 Missing docstring in public module
tests/test_auth.py:1:1: D100 Missing docstring in public module
tests/test_register.py:1:1: D100 Missing docstring in public module
tests/test_wheel.py:1:1: D100 Missing docstring in public module
tests/test_check.py:1:1: D100 Missing docstring in public module
tests/test_commands.py:1:1: D100 Missing docstring in public module
tests/test_upload.py:1:1: D100 Missing docstring in public module
tests/test_integration.py:1:1: D100 Missing docstring in public module
tests/test_cli.py:1:1: D100 Missing docstring in public module
tests/test_main.py:1:1: D100 Missing docstring in public module
```

Deleting it *does* mitigate D104, but that's only one one error:

```
$ flake8 --extend-ignore=D1 --select=D104 twine tests
twine/__init__.py:1:1: D104 Missing docstring in public package
twine/commands/__init__.py:1:1: D104 Missing docstring in public package
tests/__init__.py:1:1: D104 Missing docstring in public package
```

Reply:
> 101 and 107 are conflicting in my opinion. We'd want one or the other

Agreed. I'd vote to ignore D107 for `__init__`.